src/tui/mod.rs

```diff
@@ -122,10 +122,18 @@ impl Tui {
         self.scroll_offset = min(self.scroll_max(), scroll_offset);
     }
 
+    fn maybe_follow(&mut self) {
+        let height = self.size.height as usize;
+        if self.scroll_offset >= self.line_count - height - 1 {
```

Comment:
Nit pick: I would add parens to make order of operations more obvious.

```suggestion
        if self.scroll_offset >= self.line_count - (height + 1) {
```